test/testhdf5.h

```diff
@@ -17,17 +17,20 @@
 #ifndef TESTHDF5_H
 #define TESTHDF5_H
 
-/* Include generic testing header also */
+/* Include generic testing header */
 #include "h5test.h"
 
+/* Include testing framework functionality */
+#include "testframe.h"
```

Comment:
Separate or ideally unify. In the meantime, both headers are included here without assuming what each header includes. `testframe.h` mostly just includes `h5test.h` as a convenience since library developers have generally assumed they can just include `h5test.h` and get everything needed. This is moving a bit more toward a test either including `testframe.h` or `h5test.h`, depending on the functionality needed, until tests are maybe made consistent at some point in the future. The general rule with these changes is the testhdf5-style tests can access functionality from both `testframe.h` and `h5test.h`, but not the other way around.